### ISSUE TITLE
Flynn: Son of Crimson is reported as available in Humble library, but it isn't?

### ISSUE THREAD

**Issue by mickoz** [JosefNemec/PlayniteExtensions#264]
**Check for existing issue**
Saw nothing obvious to me searching for opened and past with "humble", "flynn", etc. keywords.

**Describe the bug**
I saw that the game "Flynn: Son of Crimson" is associated to Humble Library in PlayNite, however the game is not in the Humble App.  I checked if I owned it through other bundle, choice, etc. on Humble, but no (I was planning to buy a new bundle so I used Playnite to see what I already have and noticed that and then went looking out of curiosity and no trace of that game in Humble).

I thought it might have been available and removed, so I tried to force update on the Humble Library to see if it removes it from the list (since no more in my library), but it stays there (I suppose it would remove it if that happen or maybe that is the source of the issue?).  I looked the Configuration Options of the "Humble" (not "Humble Keys Library"), and nothing obvious give a hint of why this apparent leak might happen.

If I try to install it, it bring me to the Humble App, but without any prompt.  I tried with another game (A Hat In Time) and it does bring me to a screen to install as expected, but not with "Flynn: Son of Crimson".

So for some reason, it appears to me that game is fetched but isn't really available or... it is available through other means that PlayNite Humble addon figure out but I am not aware of... but logically all these Humble Games Collection should be through the Humble App.

Note: I have not scanned if there are other examples like this, but I can collaborate to compare available list versus reported list if that help.

**Diagnostics ID**
30527059-c05e-464d-a4b5-1a03f55f78ee

**Comment by JosefNemec** [comment]
The game is available as part of Humble Games Collection so it's being imported because you also have "Import Humble Choice games" option enabled. 

No idea why the install action doesn't work for this specific game.

>  to see if it removes it from the list
Playnite never removes games automatically, you need to do it manually even if integration plugin stops reporting a game.

**Comment by mickoz** [comment]
Yeah, outside curiosity, it doesn't matter much in practice for me for that game, but it might be important at some point or be something configurable (I have not gave much thoughts yet).

Is there a value somewhere that indicate how the game was imported, etc.!? (Like if I want to debug that the game was added on X date or whatever -- or if I would like to debug what an addon do, like list of games it fetch [yeah, I guess I could go on, understand code base and it would be trivial from there supposing code base is not hard to understand, etc.])

When I talk about fast thoughts:
- ~If I decide that game X should not appear in my library from Humble library, and if I remove and re-synch it will re-add, maybe there should be a way to ignore the importation of a specific title for a given library (if that doesn't already exist)~
// OK I just tried to remove the game and it gives me the option to add to exclusion list, so already been thought.  I however just deleted and re-imported and the game is not re-imported back as you said (make sense).

- Maybe for some library I might want to enable "erase if doesn't exist"... (or maybe it can be a kind of warning -- but I can see this being cumbersome -- just thinking out loud) -- Of course here since I personally have not customized anything, I can probably just delete all "Humble" related games, run sync, and I will have a fresh valid list for current day. (Of course if I decide to alter some meta-data, etc. then I would need to keep delta and be able to re-apply, more complicated somehow if we can't easily manipulate PlayNite database like a developer, or maybe play with tags/category if I really care about such acccuracy/refreshment -- probably hard to automatize/parametrize without some more thoughts)
  - Of course, if for some reason I altered game data (especially with Humble where we could download some DRM-free game, game might still be valid in library).  Of course for some service (Xbox game pass, etc.?) where games library might cycle, someone might want to know which one are active or past title associated, etc. (and maybe that has already been thought about!)

All in all, if that is the explanation, I would say this is "by design" and a non-issue then (I have never added game manually so far, and if we can find out that game was indeed in Humble (Trove?) then we will be 100% sure, but I think we can safely assume that is what happened here!).  So I close this issue (feel free to reply to my thoughts if you have something to say!)

Thanks

P.S., Game Id was flynn_sonofcrimson_trove (Database Id: d0bd22ff-d910-4a34-bda0-6db5128c4d51) -- probably come from that old Humble Bundle Trove (but I am not subscribed to that since long and did not give it much attention)
